docs/scarpet/api/Entities.md

```diff
@@ -698,6 +710,21 @@ Will set entity on fire for `ticks` ticks. Set to 0 to extinguish.
 
 Modifies directly player raw hunger components. Has no effect on non-players
 
+### `modify(e, 'absorption', value)`
+
+Sets the absorption value for the player. Each point is half a yellow heart.
+
+### `modify(e, 'score', value)`
+
+Sets the score of a player. Doesn't affect player xp, and is displayed upon death.
+
+### `modify(e, 'add_xp', value)`
```

Comment:
this exposes underlying implementation, which is not necessarily a good thing. `modify` should in general 'set' stuff unless its not practical. With modify(p, 'xp') or 'xp_level' you could move them back and forth.

Reply:
thing is, I tried that for wherever I could. In many cases I simply coulndt

Reply:
like there is no function for that.

Reply:
I got that logic, and realised it wasn't practical, except in the case of `score`, which is why I added that.

Reply:
Ok, I checked the XP malarchy that is in vanilla.
What makes sense to me - points are little irrelevant. We should have
for setters:
'add_xp' - nonnegative? - maybe could get it to work with negatives as well, since seems like addExperience can handle negative numbers as well, adds xp points the way vanilla `/xp add X N points` works (will change progress and level, and score)
'xp_level' - sets a level value directly, does what `/xp set X N levels`
'xp_progress' - sets directly player.experienceProgress, requires float from 0 to 1
'xp_score' - directly mess up with the score values. (using setScore)

for getters:

rename `xp_points` to `xp_progress` since it returns a float from 0 to 1

What would be helpful was to add `experience_points(level)`  (separate function in Auxiliary) that rips of getNextLevelExperience for custom input level (you will need to copy-paste it, with a reference for the future)